src/DefifaScoreCardVerifier.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity ^0.8.15;
+
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "murky/Merkle.sol";
+import "./interfaces/IDefifaScoreCardVerifier.sol";
+
+
+/**
+Custom Errors
+*/
+error INVALID_SCORECARD();
+
+//TODO: We should transfer the ownership of the verifier to the governer
+contract DefifaScoreCardVerifier is IDefifaScoreCardVerifier, Merkle {
+    // for handling precision
+    uint256 MAX_TOTAL_REDEMPTION_PERCENT = 10**6;
+    
+    /**
+    @notice Generates merkel root based on the raw scorecard data passed.
+    @param _scorecards array of the scorcard struct.
+    */
+    function generateRoot(ScoreCard[] memory _scorecards) external view override returns(bytes32) {   
+        uint256 totalRedemptionPercent;
+        bytes32[] memory data = new bytes32[](_scorecards.length);
+        // generate the merkle proof.
+        for (uint256 i = 0; i < _scorecards.length; ) {
+            data[i] = keccak256(abi.encodePacked(_scorecards[i].tierID, _scorecards[i].redemptionPercent));
+            totalRedemptionPercent += _scorecards[i].redemptionPercent;
```

Comment:
nit: If you change the `redemptionPercent` to something thats a lot smaller (ex. `uint24`) then you can make this unchecked as it can't realistically overflow